sdk/communication/Azure.Communication.Common/tests/Pipeline/PolicyTestBase.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using NUnit.Framework;
+
+namespace Azure.Communication.Pipeline
+{
+    public abstract class PolicyTestBase
```

Comment:
We should be reuse the type from Azure core test, but worth inheriting from and adding overloads that we can pass a content to ensure we can handle none-ascii encoding and escaped characters (feel free to do the later in a separate PR)

Reply:
The original `PolicyTestBase` has an assert
`Assert.IsInstanceOf<HttpPipelineSynchronousPolicy>(policy, "Use SyncAsyncPolicyTestBase base type for non-sync policies");`
HMACAuthenticationPolicy didn't meet this requirement so I thought I would make our own version